classes/Tutor.php

```diff
@@ -837,13 +856,134 @@ public static function create_database() {
 			PRIMARY KEY (withdraw_id)
 		) $charset_collate;";
 
+		$orders_table = "CREATE TABLE {$wpdb->prefix}tutor_orders (
+			id BIGINT(20) UNSIGNED AUTO_INCREMENT,
+			payment_id BIGINT(20) UNSIGNED COMMENT 'Unique payment id from payment gateway',
+			user_id BIGINT(20) UNSIGNED NOT NULL,
+			status ENUM('pending', 'completed', 'cancelled', 'refunded', 'partially-refunded') NOT NULL,
+			sub_total_price DECIMAL(13, 2) NOT NULL, -- price calculation based on course sale price
+			total_price DECIMAL(13, 2) NOT NULL,
+			order_price DECIMAL(13, 2) NOT NULL,
+			coupon_code VARCHAR(255),
+			discount_type ENUM('percentage', 'flat'),
+			discount_amount DECIMAL(13, 2),
+			discount_reason TEXT,
+			tax DECIMAL(13, 2) COMMENT 'Tax percentage',
+			tax_amount DECIMAL(13, 2),
+			fees DECIMAL(13, 2),
+			earnings DECIMAL(13, 2),
+			payment_method VARCHAR(255),
+			payment_payloads LONGTEXT,
+			note TEXT,
+			created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+			PRIMARY KEY (id),
+			KEY user_id (user_id),
+			KEY payment_id (payment_id),
+			KEY status (status)
+		) $charset_collate;";
+
+		$order_meta_table = "CREATE TABLE {$wpdb->prefix}tutor_ordermeta (
+			id BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
+			order_id BIGINT(20) UNSIGNED NOT NULL,
+			meta_key VARCHAR(255) NOT NULL,
+			meta_value LONGTEXT NOT NULL,
+			PRIMARY KEY (id),
+			KEY order_id (order_id),
+			KEY meta_key (meta_key),
+			CONSTRAINT fk_tutor_ordermeta_order_id FOREIGN KEY (order_id) REFERENCES {$wpdb->prefix}tutor_orders(id) ON DELETE CASCADE
+		) $charset_collate;";
+
+		$order_items_table = "CREATE TABLE {$wpdb->prefix}tutor_order_items (
+			id BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
+			order_id BIGINT(20) UNSIGNED NOT NULL,
+			user_id BIGINT(20) UNSIGNED NOT NULL,
+			course_id BIGINT(20) UNSIGNED NOT NULL,
+			regular_price DECIMAL(13, 2) NOT NULL, -- course regular price
+			sale_price DECIMAL(13, 2) NULL, -- course sale price
+			PRIMARY KEY (id),
+			KEY order_id (order_id),
+			KEY course_id (course_id),
+			CONSTRAINT fk_tutor_order_item_order_id FOREIGN KEY (order_id) REFERENCES {$wpdb->prefix}tutor_orders(id) ON DELETE CASCADE
+		) $charset_collate;";
+
+		$coupons_table = "CREATE TABLE {$wpdb->prefix}tutor_coupons (
+			id BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
+			coupon_code VARCHAR(50) NOT NULL,
+			coupon_title VARCHAR(255) NOT NULL,
+			coupon_description TEXT,
```

Comment:
I believe this is redundant to use `coupon_` prefix of the column names inside the `coupons` table.

Reply:
I believe, despite the table name, column names should be descriptive & it's the convention. For ex: on the wp_posts table there columns lie: post_title, post_content etc.